File: bindings/java/sources-java/info/yeppp/CpuCycleCounterState.java
/*
 *                      Yeppp! library implementation
 *
 * This file is part of Yeppp! library and licensed under the New BSD license.
 * See LICENSE.txt for the full text of the license.
 */

package info.yeppp;

/**
 * The state of the processor cycle counter.
 * <p>
 * This class is intended to use only through {@link Library#acquireCycleCounter Library.acquireCycleCounter} and {@link Library#releaseCycleCounter Library.releaseCycleCounter} methods.
 *
 * @see	Library#acquireCycleCounter
 * @see	Library#releaseCycleCounter
 */
public final class CpuCycleCounterState {

	protected CpuCycleCounterState(long state) {
		this.state = state;
	}

	protected long state;

};

File: bindings/clr/sources-csharp/library/CpuCycleCounterState.cs
/*
 *                      Yeppp! library implementation
 *
 * This file is part of Yeppp! library and licensed under the New BSD license.
 * See library/LICENSE.txt for the full text of the license.
 */

namespace Yeppp
{

	/// <summary>The state of the processor cycle counter.</summary>
	/// <remarks>This class is intended for use only through <see cref="Library.AcquireCycleCounter" /> and <see cref="Library.ReleaseCycleCounter" /> methods.</remarks>
	/// <seealso cref="Library.AcquireCycleCounter" />
	/// <seealso cref="Library.ReleaseCycleCounter" />
	public sealed class CpuCycleCounterState
	{

		internal CpuCycleCounterState(ulong state)
		{
			this.state = state;
		}

		~CpuCycleCounterState()
		{
			if (this.IsValid)
			{
				Library.ReleaseCycleCounter(this);
			}
		}

		public bool IsValid
		{
			get
			{
				return this.state != 0;
			}
		}

		internal ulong state;

	}

}
